ceremonies/meetup-validation/src/lib.rs

```diff
@@ -0,0 +1,253 @@
+	for i in participants {
+		match participant_votes[*i] {
+			v if v > 0 => continue,
+			_ => excluded_participants.push(*i),
+		}
+	}
+	excluded_participants
+		.into_iter()
+		.map(|p| (p, ExclusionReason::NoVote))
+		.collect()
+}
+
+fn get_excluded_participants_wrong_vote(
+	participants: &Vec<usize>,
+	participant_votes: &Vec<u32>,
+	n_confirmed: u32,
+) -> Vec<(usize, ExclusionReason)> {
+	let mut excluded_participants: Vec<usize> = vec![];
+	for i in participants {
+		if participant_votes[*i] != n_confirmed {
+			excluded_participants.push(*i)
+		}
+	}
+	excluded_participants
+		.into_iter()
+		.map(|p| (p, ExclusionReason::WrongVote))
+		.collect()
+}
+
+/// The algorithm works as follows:
+/// We find the participant with the fewest incoming/outgoing attestations
+/// If it if below the threhsold, we exclude the participant
+fn get_excluded_participants_num_attestations(
+	participants: &Vec<usize>,
+	participant_attestations: &Vec<Vec<usize>>,
+	threshold_fn: fn(usize) -> usize,
+) -> Vec<(usize, ExclusionReason)> {
+	let mut relevant_attestations = filter_attestations(participants, participant_attestations);
+
+	let mut excluded_participants: Vec<(usize, ExclusionReason)> = vec![];
+	let mut participants_to_process: Vec<usize> = participants.clone();
+
+	while participants_to_process.len() > 0 {
+		let (
+			participants_grouped_by_outgoing_attestations,
+			participants_grouped_by_incoming_attestations,
+		) = group_participants_by_num_incoming_and_outgoing_attestations(
+			&participants_to_process,
+			&relevant_attestations,
+		);
+
+		let min_num_outgoing_attestations = participants_grouped_by_outgoing_attestations[0].0;
+		let min_num_incoming_attestations = participants_grouped_by_incoming_attestations[0].0;
+
+		if min_num_incoming_attestations < min_num_outgoing_attestations {
+			if min_num_incoming_attestations < threshold_fn(participants_to_process.len()) {
+				let ps = participants_grouped_by_incoming_attestations[0].1.clone();
+				ps.clone().into_iter().for_each(|p| {
+					excluded_participants.push((p, ExclusionReason::TooFewIncomingAttestations))
+				});
+
+				// remove the participants from the included participants and the attestation vectors
+				participants_to_process.retain(|k| !ps.contains(k));
+				relevant_attestations =
+					filter_attestations(&participants_to_process, &relevant_attestations);
+				continue
+			}
+		} else {
+			if min_num_outgoing_attestations < threshold_fn(participants_to_process.len()) {
+				let ps = participants_grouped_by_outgoing_attestations[0].1.clone();
+				ps.clone().into_iter().for_each(|p| {
+					excluded_participants.push((p, ExclusionReason::TooFewOutgoingAttestations))
+				});
+
+				// remove the participants from the included participants and the attestation vectors
+				participants_to_process.retain(|k| !ps.contains(k));
+				relevant_attestations =
+					filter_attestations(&participants_to_process, &relevant_attestations);
+				continue
+			}
```

Comment:
The if and else statement here are identical except for the exclusion reason, you should either define a function or a closure to remove the redundant definition